guest/prefilterextensions/prefilterextensions.go

```diff
@@ -0,0 +1,76 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package prefilterextensions exports an api.PreFilterExtensions to the host. Only import this
+// package when setting Plugin, as doing otherwise will cause overhead.
+
+package prefilterextensions
+
+import (
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/filter"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/cyclestate"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/imports"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/plugin"
+)
+
+// prefilterextensions is the current plugin assigned with SetPlugin.
+var prefilterextensions api.PreFilterExtensions
+
+// SetPlugin is exposed to prevent package cycles.
+func SetPlugin(preFilterExtensions api.PreFilterExtensions) {
+	if preFilterExtensions == nil {
+		panic("nil preFilterExtensions")
+	}
+
+	plugin.MustSet(preFilterExtensions)
+}
+
+// prevent unused lint errors (lint is run with normal go).
+var (
+	_ func() uint32 = _addpod
+	_ func() uint32 = _removepod
+)
+
+// _addPod is only exported to the host.
+//
+// export addpod
+func _addpod() uint32 { //nolint
+	if prefilterextensions == nil { // Then, the user didn't define one.
+		// Unlike most plugins we always export reserve so that we can reset
+		// the cycle state: return success to avoid no-op overhead.
+		return 0
+	}
+
+	status := prefilterextensions.AddPod(cyclestate.Values, cyclestate.Pod, cyclestate.Pod, &filter.NodeInfo{})
```

Comment:
`cyclestate.Pod` is `podToSchedule`, which is currently being scheduled, whilst `podInfoToAdd` is a completely different Pod.

https://github.com/kubernetes/kubernetes/blob/2363cdcc399cbf428210efb2c51575ddcad2b84a/pkg/scheduler/framework/interface.go#L389

So, you have to get a Pod equivalent to `podInfoToAdd` from the scheduler side thru wasm runtime.

Reply:
@sanposhiho 
Thank you for your feedback.For getting a Pod equivalent to podInfoToAdd from `wasm runtime` , is there any implementation (or documentation) that I can refer...?

Reply:
I missed this message, sorry!

You can refer to how [`nodeToStatus`](https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/blob/main/guest/postfilter/postfilter.go#L87-L100) is implemented in https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension/pull/75/files.